src/pages/pattern/pattern-details/pattern-details.js

```diff
@@ -0,0 +1,288 @@
+      label: 'зроблений вручну',
+      handler: (e) => setHandmade(e.target.checked)
+    },
+    {
+      id: 'available',
+      value: available,
+      checked: available,
+      color: 'primary',
+      label: 'доступний',
+      handler: (e) => setAvailable(e.target.checked)
+    }
+  ];
+
+  const checkboxInputs = checkboxes.map(
+    ({ color, checked, label, value, handler }) => (
+      <FormControlLabel
+        key={label}
+        value={value}
+        checked={checked}
+        control={<Checkbox color={color} />}
+        label={label}
+        labelPlacement='start'
+        onChange={handler}
+      />
+    )
+  );
+
+  return (
+    <div className={classes.detailsContainer}>
+      <form className={classes.form} onSubmit={patternSaveHandler}>
+        <FormControl className={classes.patternDetails}>
+          <Grid container spacing={1}>
+            {checkboxInputs}
+            <Grid item xs={12}>
+              <Paper className={classes.patternItemUpdate}>
+                <TextField
+                  id='large'
+                  className={classes.textField}
+                  variant='outlined'
+                  label='Фото великого розміру'
+                  multiline
+                  value={large}
+                  onChange={(e) => setLarge(e.target.value)}
+                  required
+                />
+                <TextField
+                  id='medium'
+                  className={classes.textField}
+                  variant='outlined'
+                  label='Фото середнього розміру'
+                  multiline
+                  value={medium}
+                  onChange={(e) => setMedium(e.target.value)}
+                  required
+                />
+                <TextField
+                  id='small'
+                  className={classes.textField}
+                  variant='outlined'
+                  label='Фото малого розміру'
+                  multiline
+                  value={small}
+                  onChange={(e) => setSmall(e.target.value)}
+                  required
+                />
+                <TextField
+                  id='thumbnail'
+                  className={classes.textField}
+                  variant='outlined'
+                  label='Фото найменшого розміру'
+                  multiline
+                  value={thumbnail}
+                  onChange={(e) => setThumbnail(e.target.value)}
+                  required
+                />
+                <TextField
+                  id='material'
+                  className={classes.textField}
+                  variant='outlined'
+                  label='Матеріал'
+                  multiline
+                  value={material}
+                  onChange={(e) => setMaterial(e.target.value)}
+                  required
+                />
+              </Paper>
+            </Grid>
+            <Grid item xs={6}>
+              <Paper className={classes.patternItemUpdate}>
```

Comment:
Please, move this component `Paper` and its inner components to the separate component to reduce size of current file and improve readability

Reply:
refactored

Reply:
fixed